### ISSUE TITLE
Secondary logins display is inconsistent

### ISSUE THREAD

**Issue by NikkiWines** [Expensify/App#13350]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log into a newDot Expensify account with multiple secondary logins 
2. Navigate to Settings > Profile 
3. Confirm the logins display match your primary logins as shown in oldDot 

## Expected Result:
Your primary login in oldDot should be displayed in newDot and at least one secondary login of the opposite type (e.g. if primary is an email, then a phone number) should display (if there was one) 

## Actual Result:
Occasionally, the results do not align with what is shown in oldDot 

| NewDot  | OldDot |
| ------------- | ------------- |
| ![bbe976ac-648c-470a-84f1-8ca41f290c9c](https://user-images.githubusercontent.com/16747822/205775698-da634011-3c85-4ba4-b2dd-a7e0f7ff1bd0.jpg) | ![Screenshot 2022-12-05 at 16 26 20](https://user-images.githubusercontent.com/16747822/205775992-a83b4bd2-a4fd-4827-9558-1f0a9a126461.png) |
| <img width="378" alt="screenshot_2022-12-05_at_11 52 27_am__1_" src="https://user-images.githubusercontent.com/16747822/205775700-dd3ff403-1cfb-4287-9253-8c1ac159027d.png"> | ![Screenshot 2022-12-05 at 15 58 46](https://user-images.githubusercontent.com/16747822/205775988-8b84694a-e645-4ec4-a241-b2d24c973449.png) |

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** v1.2.36-1
**Reproducible in staging?:**  yes 
**Reproducible in production?:**  yes
**Email or phone of affected tester (no customers):** nikkiwines@expensify.com, joanie@expensify.com 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
**Expensify/Expensify Issue URL:** N/A
**Issue reported by:** @joaniew 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1670271497760679

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by NikkiWines** [comment]
At first glance, I think a contributing factor could be that we're only ever pulling the login list when the user navigates to newDot via short-lived authtoken (via [`User.getUserDetails()`](https://github.com/Expensify/App/blob/d8f64c2d060847380bc14c91fbf7b6bc9c8e9a58/src/libs/actions/User.js#L95) called [here](https://github.com/Expensify/App/blob/09e3bf19e78b69e807aed54675ef0f2e26522f21/src/libs/actions/Session/index.js)) or if they set a new secondary login in the product (via [`User.setSecondaryLoginAndNavigate()`](https://github.com/Expensify/App/blob/d8f64c2d060847380bc14c91fbf7b6bc9c8e9a58/src/libs/actions/User.js#L174) called [here](https://github.com/Expensify/App/blob/d9e523c7e8ab40fcf75419c59f101af42a915b0d/src/pages/settings/AddSecondaryLoginPage.js#L77))


If the user updates their secondary login in oldDot that won't be reflected in newDot unless one of the two other aforementioned actions occur. 

I also thought the issue might be that we're accidentally filtering out logins that are present across two partners (i.e. Expensify.com and google sign in) but some local testing confirms that's not the case.

**Comment by NikkiWines** [comment]
Nevermind it is exactly that, we're filtering out emails when we call `$user->getLoginListAssociative()` in Web [here](https://github.com/Expensify/Web-Expensify/blob/3738adf7b7149dcc691e8ef056089cc90d597713/lib/UserAPI.php#L984-L989)

![image](https://user-images.githubusercontent.com/16747822/205787403-d6b727ec-3c20-4c6e-b001-079743ca3bb7.png)

**Comment by Beamanator** [comment]
Looks like a dupe of: https://github.com/Expensify/App/issues/13131

I have a PR being reviewed too: https://github.com/Expensify/Web-Expensify/pull/35704 😬

**Comment by abekkala** [comment]
Nice so it looks like there is already a fix in the works for this!

**Comment by abekkala** [comment]
What would you like from the BZ team here (essentially me)?
Do you need an Upwork job created? If so, what exactly is the job? 
Or is this not external and will remain Internal?

**Comment by NikkiWines** [comment]
This is internal and not reported by a contributor so I think there's not too much to do on the BZ side of things.

**Comment by JmillsExpensify** [comment]
Agreed, especially since the PR is Web-Expensify and that's a private repo.

**Comment by abekkala** [comment]
Ok, so we wouldn't even have a contributor Review the PR? 
I'm trying to pinpoint what action I need to take here as the assignee

**Comment by NikkiWines** [comment]
At the moment no action is needed from you here.

**Comment by NikkiWines** [comment]
No overdue, got a PR in review!

**Comment by NikkiWines** [comment]
^^ same update as above - hoping to get the PR reviewed shortly

**Comment by NikkiWines** [comment]
PR has been merged! Will close this once the PR has been deployed but I think for now it can be a weekly

**Comment by abekkala** [comment]
I'm now ooo and will need to reassign.
@NikkiWines can you confirm the only thing here for the BZ team member is the checklist?  It looks like PR reviews were also done internally so I don't think there is a contributor payment needed anywhere, correct?

**Comment by joekaufmanexpensify** [comment]
Looks like PR was deployed to prod a few days ago. Given it was reviewed internally, all we need is BZ checklist. I expect it was not posted since the fix wasn't in the App repo. I'll manually create it.

**Comment by joekaufmanexpensify** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ]  [@NikkiWines] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@NikkiWines] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@NikkiWines] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@joekaufmanexpensify] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:

**Comment by NikkiWines** [comment]
@joekaufmanexpensify this issue ended up being a duplicate of https://github.com/Expensify/App/issues/13131, and my PR ended up just adjusting [a recently merged PR](https://github.com/Expensify/Web-Expensify/pull/35704) that had already been merged to resolve that issue.

Given that, I think https://github.com/Expensify/App/issues/13131 should have the BZ checklist, not this issue 🤔 thoughts?

**Comment by joekaufmanexpensify** [comment]
Sounds good to me! I'll post it there.

**Comment by joekaufmanexpensify** [comment]
Re-opened other issue and added BZ checklist there. Closing this one as PR is on prod, and there is no payment to issue!
